eng/pipelines/templates/steps/analyze.yml

```
@@ -24,6 +24,16 @@ steps:
     failOnStderr: false
     workingDirectory: '${{parameters.GoWorkspace}}'
 
+  - pwsh: |
+      $modDirs = (./eng/scripts/get_module_dirs.ps1 -serviceDir $(SCOPE))
+      foreach ($md in $modDirs) {
+        pushd $md
+        Get-ChildItem doc.go
+        if ($LASTEXITCODE) { exit $LASTEXITCODE }
+      }
```

`$LASTEXITCODE` only applies to exit codes from external process calls. Powershell commands/functions are handled via `throw`/exceptions. You could also use a `Test-Path` conditional here.

```suggestion
      foreach ($md in $modDirs) {
        Get-ChildItem $md/doc.go
      }